openpgp/packet/private_key_test.go

```diff
@@ -56,6 +65,77 @@ func TestPrivateKeyRead(t *testing.T) {
 	}
 }
 
+func TestPrivateKeyEncrypt(t *testing.T) {
+	var encryptedMsg, decryptedMsg []byte
+	var encryptedC1, encryptedC2 *big.Int
+
+	for i, test := range privateKeyTests {
+		packet, err := Read(readerFromHex(test.privateKeyHex))
+		if err != nil {
+			t.Errorf("#%d: failed to parse: %s", i, err)
+			continue
+		}
+		privKey := packet.(*PrivateKey)
+		err = privKey.Decrypt([]byte("testing"))
+		if err != nil {
+			t.Errorf("#%d: failed to decrypt: %s", i, err)
+			continue
+		}
+
+		switch pubKey := privKey.PublicKey.PublicKey.(type) {
```

Comment:
You're probably right, we are mostly concerned that it can be serialized and deserialized when encrypted, and if it can be decrypted. There are other tests responsible for message encryption. However I like the idea of trying to decrypt a message to see if we end up with functional key after passphrase decryption.